src/Harmony/Signer.cpp

```diff
@@ -26,40 +26,280 @@ Signer::sign(const uint256_t &chainID, const PrivateKey &privateKey, const Data
     return values(chainID, signature);
 }
 
-Proto::SigningOutput Signer::sign(const TW::Harmony::Proto::SigningInput &input) noexcept {
+template <typename T>
+Proto::SigningOutput Signer::prepareOutput(const Data &encoded, const T &transaction) noexcept {
+    auto protoOutput = Proto::SigningOutput();
+
+    auto v = store(transaction.v);
+    auto r = store(transaction.r);
+    auto s = store(transaction.s);
+
+    protoOutput.set_encoded(encoded.data(), encoded.size());
+    protoOutput.set_v(v.data(), v.size());
+    protoOutput.set_r(r.data(), r.size());
+    protoOutput.set_s(s.data(), s.size());
+
+    return protoOutput;
+}
+
+template <>
+Proto::SigningOutput
+Signer::sign<Transaction>(const TW::Harmony::Proto::SigningInput &input) noexcept {
     auto key = PrivateKey(Data(input.private_key().begin(), input.private_key().end()));
     Address toAddr;
-    if (!Address::decode(input.to_address(), toAddr)) {
+    if (!Address::decode(input.transaction_message().to_address(), toAddr)) {
         // invalid to address
         return Proto::SigningOutput();
     }
     auto transaction = Transaction(
```

Comment:
Remove template and handle different messages in `Transaction` class

Reply:
@hewigovens you mean ```Signer``` class?